### PR TITLE
[TICKET 2] :rocket: update project to nextjs13

### PR DESCRIPTION
**Pull request by czhoffmann23**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
Project has to be updated to NextJs 13
## What is the new behavior?
- Resolve package dependencies.
- Restructure providers according to NextJs13 implementation.
- Removed `_app.tsx` - `document.tsx` and moved to `layout.tsx`.
- Restructure pages to work with `src/app/* ` and prefix `[route]/page.tsx`.
- Update Packages: `chakra ui`, `typescript`,  `nextjs`, `@next/eslint-plugin-next`, `react` and `react-dom`.
- Updates in API call, cause api ` api spacex land graphql` do not work.
- Add encoding package according to terminal warning/issue: [GITHUB ISSUE](https://github.com/supabase/supabase-js/issues/612).
## How can your PR break the app?
Super breaking changes
## Other information
TODO:
- `[...nextauth].ts` must be fixed and tested properly.
- `NextAuth` with `Github` need some secrets and ids to be implemented.

Everything else is now working properly.

PD1: `NEXT_PUBLIC_URL` in Vercel is pointing to -> `https://rickandmortyapi.com/graphql`
PD2: Vercel project `node` version was updated to `18.x`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +50,7 @@
     "husky": "^7.0.2",
     "lint-staged": "^11.1.2",
     "prettier": "^2.4.1",
-    "typescript": "^4.4.3"
+    "typescript": "^4.5.2"
```
Comment (Yhozen) [inline package.json]: not bleeding edge enough :-1:

Comment (Yhozen) [reply package.json]: not evil- @tomcastro  enough

File: src/context/ReactQueryCustomContext.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { QueryClientProvider } from 'react-query'
+
+import { queryClient } from 'lib/queryClient'
+
+export const ReactQueryCustomContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode
+}) => {
+  const [client] = useState(queryClient)
```
Comment (Byevenes) [inline src/context/ReactQueryCustomContext.tsx]: suggestion for configuring queryClient globally
`  const [newClient] = useState(() => new QueryClient({
    queries: {
     retry: 2,
   },
 queryCache: new QueryCache({
   onError: (error, query) => {
     if (query.state.data !== undefined) {
       toast.error(error.message)
     }
   },
 }),
})) `

- 🎉 only show error toasts if we already have data in the cache which indicates a failed background update.
- retry 3 times before marking the query as an errory.
- etc ...

File: package.json [hunk 3]
```diff
@@ -48,7 +50,7 @@
     "husky": "^7.0.2",
     "lint-staged": "^11.1.2",
     "prettier": "^2.4.1",
-    "typescript": "^4.4.3"
+    "typescript": "^5.0.4"
```
Comment (Yhozen) [inline package.json:53]: :fire:

File: package.json [hunk 4]
```diff
@@ -21,25 +22,26 @@
...
     "@types/jsonwebtoken": "^8.5.5",
     "@types/node": "14.14.35",
-    "@types/react": "^17.0.21",
-    "@types/react-dom": "^17.0.9",
+    "@types/react": "^18.1.0",
+    "@types/react-dom": "^18.1.0",
     "@types/react-table": "^7.7.3",
     "@typescript-eslint/eslint-plugin": "^4.31.1",
     "@typescript-eslint/parser": "^4.31.1",
+    "encoding": "^0.1.13",
```
Comment (Yhozen) [inline package.json:42]: is it necessary?

**Review by Yhozen — APPROVED** [review]
let's do a follow up PR with more features :money_mouth_face:

File: package.json (hunk 4 above)
Comment (czhoffmann23) [reply package.json:42]: [GITHUB ISSUE](https://github.com/supabase/supabase-js/issues/612)

**Review by Byevenes — APPROVED** [review]

**Pull request merged**